website/addons/osfstorage/routes.py

```diff
@@ -112,6 +112,61 @@
             views.osfstorage_copy_hook,
             json_renderer,
         ),
+
+        Rule(
```

Comment:
I'm not a big fan of these as they aren't very restful. Also, I'm a little concerned about the use of a rent metaphor in general because this kind of service has been around forever and I don't recall it ever having been considered a "rental" rather than, say, a "checkout". If there's precedence, perhaps, but if it's new just to be new, I'd say go with the existing metaphor. Alternately, if this does something materially different than a checkout, perhaps.

As far as the URLs, they aren't particularly RESTful. They should be collections that you GET, POST, and DELETE. So, for example, `/project/<pid>/osfstorage/{fid}/rentals` would POST to rent, DELETE to return, and GET to get the status. A force return, I'm expecting, would be the same thing as a return except being done by someone who isn't the renter, so there's no need for an alternate endpoint. Take out the `{fid}` to POST a rent all and DELETE to return all.

Anyways, mostly small changes with perhaps one refactor should be great.